server/src/apub/community.rs

```diff
@@ -50,6 +55,66 @@ impl ActorType for Community {
   fn public_key(&self) -> String {
     self.public_key.to_owned().unwrap()
   }
+
+  /// As a local community, accept the follow request from a remote user.
+  fn send_accept_follow(&self, follow: &Follow) -> Result<(), Error> {
+    let actor_uri = follow
+      .follow_props
+      .get_actor_xsd_any_uri()
+      .unwrap()
+      .to_string();
+
+    let mut accept = Accept::new();
+    accept
+      .object_props
+      .set_context_xsd_any_uri(context())?
+      // TODO: needs proper id
+      .set_id(
+        follow
+          .follow_props
+          .get_actor_xsd_any_uri()
+          .unwrap()
+          .to_string(),
+      )?;
+    accept
+      .accept_props
+      .set_actor_xsd_any_uri(self.actor_id.to_owned())?
+      .set_object_base_box(BaseBox::from_concrete(follow.clone())?)?;
+    let to = format!("{}/inbox", actor_uri);
+    send_activity(
+      &accept,
+      &self.private_key.to_owned().unwrap(),
+      &self.actor_id,
+      vec![to],
+    )?;
+    Ok(())
+  }
+
+  /// For a given community, returns the inboxes of all followers.
+  fn get_follower_inboxes(&self, conn: &PgConnection) -> Result<Vec<String>, Error> {
+    debug!("got here.");
+
+    Ok(
+      CommunityFollowerView::for_community(conn, self.id)?
+        .into_iter()
+        // TODO eventually this will have to use the inbox or shared_inbox column, meaning that view
+        // will have to change
+        .map(|c| {
+          // If the user is local, but the community isn't, get the community shared inbox
+          // and vice versa
+          if c.user_local && !c.community_local {
+            get_shared_inbox(&c.community_actor_id)
+          } else if !c.user_local && c.community_local {
+            get_shared_inbox(&c.user_actor_id)
+          } else {
+            "".to_string()
+          }
```

Comment:
This is some really weird logic, but its the only way I could get it to work.

If you only use the `community_actor_id`, then a federated user in the `CommunityFollower` table will still have local `community_actor_id`, since that community is local, but the user isn't, and it will think that the only place it needs to send a post, is to itself.

And the same problem exists for a local user and a federated community.

Reply:
If there's not shared inbox, you can fall back on the normal inbox

Reply:
So basically, if a local user creates a post in a remote community, this code will send it only to the inbox of the community? That seems correct, at least once we have communities `Announce` those incoming posts. But it really should be handled somewhere else. You dont have to change it in this PR though.

If you cant get an inbox (not sure when that would happen), you should probably throw an error.

Reply:
> if a local user creates a post in a remote community, this code will send it only to the inbox of the community?

Yep... I'll pry need some help with the announce stuff.

Reply:
No problem I can implement that.